ihp-hsx/IHP/HSX/UncheckedHSX.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE TemplateHaskell, QuasiQuotes, OverloadedStrings #-}
+module IHP.HSX.UncheckedHSX (uncheckedHsx) where
```

Comment:
This introduces lot's of redundant code.

I'd suggest to rather extend the `quoteHsxExpression` function with an argument like `checkMarkup :: Bool`.

So turning:
```haskell
hsx :: QuasiQuoter
hsx = QuasiQuoter {
        quoteExp = quoteHsxExpression,
        quotePat = error "quotePat: not defined",
        quoteDec = error "quoteDec: not defined",
        quoteType = error "quoteType: not defined"
    }

```

into:

```
hsxWithOptions :: Bool -> QuasiQuoter
hsxWithOptions checkMarkup = QuasiQuoter {
        quoteExp = quoteHsxExpression checkMarkup,
        quotePat = error "quotePat: not defined",
        quoteDec = error "quoteDec: not defined",
        quoteType = error "quoteType: not defined"
    }

hsx :: QuasiQuoter
hsx = hsxWithOptions True

uncheckedHsx :: QuasiQuoter
uncheckedHsx = hsxWithOptions False
```

Then that boolean flag needs to be passed down to the parser and we'd need to add an `if checkMarkup` at all places where the tags are checked against the whitelist

Reply:
Thank you. I agree, I shot from the hip to start somewhere :) I will try what you suggest